### PR TITLE
Keep version of used components and their version at a single yaml file

### PR DESCRIPTION
**Pull request by RamLavi**
In order to make U/S->D/S automation possible and clearer which versions are we using, we will keep them in a dedicated yaml. This file should be machine-readable, and must be updated every time a supported version of a component is changed.

Signed-off-by: Ram Lavi <ralavi@redhat.com>

### CODE REVIEW

File: pkg/components/componenets_config.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+componenets:
```
Comment (phoracek) [inline pkg/components/componenets_config.yaml]: Let's keep it in the root of the project as components.yaml

File: pkg/components/componenets_config.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  multus:
+    componentName: "multus"
+    Version: "0.2.0"
```
Comment (phoracek) [inline pkg/components/componenets_config.yaml]: Use lower case "v".

Also, some of the tags start with "v", not the number alone. Please check.

File: pkg/components/componenets_config.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+componenets:
+  multus:
+    componentName: "multus"
```
Comment (phoracek) [inline pkg/components/componenets_config.yaml]: If we keep components under key with their name, let's drop redundant "componentName".

File: components.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+componenets:
+  multus:
+    version: "v3.4"
```
Comment (phoracek) [inline components.yaml]: In this case, we don't use 3.4 directly, we use an untagged version that came after it.

You can see it in the image tag here: https://github.com/kubevirt/cluster-network-addons-operator/blob/master/pkg/components/components.go#L22

In this case, the version should be therefore set to `bf61002b4c83fcc0ca29c271108ca6afa381998f`

Comment (RamLavi) [reply components.yaml]: is 
```version: "bf61002b4c83fcc0ca29c271108ca6afa381998f"```
 be good then? i.e., do you think future scripts will be able to interpret it in the same manner they do a specific version?

Comment (phoracek) [reply components.yaml]: We may need to add "branch" or "revision" later, but I think this is good enough for now. Let's adjust it once we have a real user and real issue.

I'm little nervous that we introduce this file and has some redundancy, but let's use it for reference and docs now, then we start basing automation (both D/S and U/S around it). We can change it whenever, it's no API, just our helper.

**Comment by phoracek** [comment]
/approve
/lgtm
/release-note-none
/override pull-e2e-cluster-network-addons-operator-lifecycle-workflow-okd
/override pull-e2e-cluster-network-addons-operator-lifecycle-workflow-k8s
/override pull-e2e-cluster-network-addons-operator-lifecycle-okd
/override pull-e2e-cluster-network-addons-operator-lifecycle-k8s

**Pull request merged**